modules/carousel/jetpack-carousel.php

```diff
@@ -540,9 +540,57 @@ class_exists( 'Jetpack_AMP_Support' )
 			 * @param array $extra_data Array of data about the site and the post.
 			 */
 			$extra_data = apply_filters( 'jp_carousel_add_data_to_container', $extra_data );
+
 			foreach ( (array) $extra_data as $data_key => $data_values ) {
-				$html = str_replace( '<div ', '<div ' . esc_attr( $data_key ) . "='" . json_encode( $data_values ) . "' ", $html );
-				$html = str_replace( '<ul class="wp-block-gallery', '<ul ' . esc_attr( $data_key ) . "='" . json_encode( $data_values ) . "' class=\"wp-block-gallery", $html );
+				// Do not go any further if DOMDocument is disabled on the server.
+				if ( ! class_exists( 'DOMDocument' ) ) {
+					return $html;
+				}
+
+				// Let's grab all containers from the HTML.
+				$dom_doc = new DOMDocument();
+
+				/*
+				 * The @ is not enough to suppress errors when dealing with libxml,
+				 * we have to tell it directly how we want to handle errors.
+				 */
+				libxml_use_internal_errors( true );
```

Comment:
`libxml_use_internal_errors()` returns the previous value, which we should store and use below.

```suggestion
				$old_libxml_use_internal_errors = libxml_use_internal_errors( true );
```